cram/cram_io.c

```diff
@@ -4575,3 +4575,27 @@ int cram_set_voption(cram_fd *fd, enum hts_fmt_option opt, va_list args) {
 
     return 0;
 }
+
+#define TEMPLATE_LEN 38
+int cram_check_EOF(cram_fd *fd)
+{
+    uint8_t buf[TEMPLATE_LEN];
+    const uint8_t template[TEMPLATE_LEN] = {
+        0x0f, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff, 0x0f, 0xe0,
```

Comment:
The data here is length (0x0f 00 00 00) and ref_seq_id (-1 == 0xff ff ff ff ff 0f).
The last 4 bits of that ref_seq_id are undefined due to the wording in the ITF8 description.  Traditionally this meant 0s for my code and 1s in the Java code, leading to 0x0f vs 0xff issues.

It looks like we now agree on this interpretation, but for a while it meant EOF blocks were being generated in two different ways.

The safest option is to AND off those 4 bits first before doing the check.